app/controllers/products_controller.rb

```diff
@@ -0,0 +1,55 @@
+class ProductsController < ApplicationController
+  def index
+    @products = params[:filters].nil? ? Product.all : filtered_products
+    @products_found = @products.count
+    @products = @products.paginate(page: params[:page])
+  end
+
+  def show
+    get_product_or_redirect
+    @related_products = Product.related_products(@product)
+                               .limit(Settings.will_paginate.per_page)
+  end
+  private
+
+  def get_product_or_redirect
```

Comment:
ko đặt lên method là get
```suggestion
  def load_product
```